core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1535,6 +1535,10 @@ abstract class RDD[T: ClassTag](
 
   private[spark] var checkpointData: Option[RDDCheckpointData[T]] = None
 
+  // Whether recursively checkpoint all RDDs that are marked with the checkpoint flag.
+  private val recursiveCheckpoint =
+    Option(sc.getLocalProperty("spark.checkpoint.recursive")).map(_.toBoolean).getOrElse(false)
```

Comment:
Well its always recursive. The difference is whether checkpoint all that have been marked or not.